### DISCUSSION TITLE
Overrideable (normal) Functions

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2862] [Suggestions]
Allow compile-time functions to be overridden with a special attribute/keyword.

This is mainly for libraries where certain functions are expected to or can be overridden by the user.

The syntax could be like so:

```typescript
override function number myFunc() {}

@override
function number myFunc() {}
```

I am unsure whether the keyword should be applied in the library or in the implementing file.

**Comment by vurvdev** [comment]
Need a really good reason as to why this should happen when we have lambdas now.

I understand overriding in terms of OOP, but this is just sloppy and would lead to functions erroring at runtime from not existing all over again..
